### PR TITLE
terminal-unix: signal related fixes

### PR DESCRIPTION
**Pull request by N-R-K**
Some signal handler related fixes. Also incorporates the `isatty` change from https://github.com/mpv-player/mpv/pull/13224.

### CODE REVIEW

File: osdep/terminal-unix.c [hunk 1]
```diff
@@ -437,10 +422,19 @@ static MP_THREAD_VOID terminal_thread(void *ptr)
...
                 do_deactivate_getch2();
-            else if (c == PIPE_CONT)
+                if (isatty(tty_out)) {
+                    (void)write(tty_out, TERM_ESC_RESTORE_CURSOR,
+                                sizeof(TERM_ESC_RESTORE_CURSOR) - 1);
+                }
+                // make do with SIGSTOP since it's effectively the same and
+                // there's no way to get a function pointer to the "real
+                // default" handler.
+                raise(SIGSTOP);
```
Comment (N-R-K) [inline osdep/terminal-unix.c:434]: There's a possible race here where if we've received a continue signal and it's queued in the pipe, we'll go to stop state before processing the cont signal. No clue how to fix it though.

But unlikely to happen in practice, and if it does happen, just sending another cont signal should fix it. So maybe not worth worrying about it (?).

**Review by N-R-K — COMMENTED** [review]
Since signal handlers are hairy, I'd like to think through this a bit more before I'm comfortable with the patch.

Opening the PR mainly for review purposes and thus marked as draft.

**Comment by N-R-K** [comment]
Other than [this](https://github.com/mpv-player/mpv/pull/13238/files#r1443796168) one possible race (which I don't see any way of fixing), this should be okay.

**Comment by N-R-K** [comment]
Changed the patch to retain printing `TERM_ESC_RESTORE_CURSOR` to `stderr` according to this discussion: https://github.com/mpv-player/mpv/pull/13224#discussion_r1451574508

File: osdep/terminal-unix.c [hunk 2]
```diff
@@ -437,10 +422,19 @@ static MP_THREAD_VOID terminal_thread(void *ptr)
...
-            if (c == PIPE_STOP)
+            if (c == PIPE_STOP) {
                 do_deactivate_getch2();
-            else if (c == PIPE_CONT)
+                if (isatty(STDERR_FILENO)) {
+                    (void)write(STDERR_FILENO, TERM_ESC_RESTORE_CURSOR,
+                                sizeof(TERM_ESC_RESTORE_CURSOR) - 1);
+                }
+                // make do with SIGSTOP since it's effectively the same and
+                // there's no way to get a function pointer to the "real
```
Comment (sfan5) [inline osdep/terminal-unix.c:432]: nit: I think those handlers actually live in the kernel so while "no way to get a function ptr" is technically correct it also doesn't make sense.
I suggest rewording this to "no way to explicitly invoke the default signal handler"

**Comment by kasper93** [comment]
> Also incorporates the isatty change from https://github.com/mpv-player/mpv/pull/13224.

@N-R-K Could you also pull the change in `mp_msg_flush_status_line`? To not split this changes into multiple prs.

File: osdep/terminal-unix.c (hunk 2 above)
Comment (N-R-K) [reply osdep/terminal-unix.c]: Done
